enhancements/network/ovs-hardware-offload.md

```diff
@@ -0,0 +1,242 @@
+to pod namespace and plugging VF representor to `br-int` OVS bridge. The
+relevant [patches](https://github.com/openshift/ovn-kubernetes/commit/6c96467d0d3e58cab05641293d1c1b75e5914795) are available downstream.
+
+#### Cluster Network Operator
+
+- Cluster Network Operator will enable hardware offload and configure tc-policy
+on node OpenvSwitch instance and restart OpenvSwitch instance for hardware
+offload change to take effect.
+
+- Enabling `hw-offload` in CNO `ovs-node` DaemonSet is equivalent to executing
+`ovs-vsctl set Open_vSwitch . other_config:hw-offload=true`. The default value
+is `false`.
+
+- Setting `tc-policy` in CNO `ovs-node` DaemonSet is equivalent to executing
+`ovs-vsctl set Open_vSwitch . other_config:tc-policy=skip_sw`. The default
+value is `none`, optional values can be one of `none`, `skip_sw` and `skip_hw`.
+This field is only relevant if `hw-offload` is enabled.
+
+#### Multus
+
+- Multus will support specifying net-attach-def with customized annotations for
+[default cluster network in pod annotations](https://github.com/intel/multus-cni/blob/master/doc/configuration.md#specify-default-cluster-network-in-pod-annotations).
+Customized annotations will contain a resource name requesting SR-IOV device.
+
+#### SR-IOV Operator
+
+- SR-IOV Operator will support new device type `switchdev` in Operator
+[SriovNetworkNodePolicy API](https://github.com/openshift/sriov-network-operator/blob/master/pkg/apis/sriovnetwork/v1/sriovnetworknodepolicy_types.go#L33).
+
+- SR-IOV config daemon (a DaemonSet managed by SR-IOV Operator) will support
+changing e-switch mode from `legacy` to `switchdev` on the PF device and
+exposing VF representor information in `SriovNetworkNodeState` status.
+
+- SR-IOV network device plugin (a Device Plugin DaemonSet managed by SR-IOV
+Operator) supports advertising VF resource to kubelet.
+
+#### Machine Config Operator (MCO) [optional]
+
+- MCO will be used to create linux bond interface on two identical PFs that
+are capable of doing OVS hardware offload and create vlan interfaces on bond
+interface. The supported bond modes are `Active-Backup`, `Active-Active`
+and `LACP`. This is not required if OVS Hardware Offload is used on PF
+device directly.
+
+### User Stories
+
+#### Story 1
+
+Workflow of creating a SR-IOV pod using OVS hardware offload:
+- [optional] Deploy a baremetal cluster with API network on bonded PFs
+- Enable OVS hardware offload for OpenvSwitch instances via CNO
+- Install SR-IOV Operator from Operator Hub
+- Provision PFs to desired number of VFs via SR-IOV Operator
+- Create a pod specifying cluster network with customized net-attach-def
+
+### Implementation Details/Notes/Constraints
+
+- When enabling OVS Hardware Offload with SR-IOV VF LAG, PFs should first be
+provisioned with desired number of VFs, then configure PFs to `switchdev`
+mode, linux bond interface shall be created after above two steps. Since linux
```

Comment:
*Who* provisions the PFs? *Who* configures them to switchdev mode? *Who* creates the bond interface? It's very unclear throughout this doc when you're saying that the administrator needs to do something vs when you're saying that OCP will do something. Use more active voice and less passive voice.

Reply:
Updated by splitting it to two points:
```
- SR-IOV Operator should first provision PFs with desired number of VFs,
then configure PFs to `switchdev` mode.

- When enabling OVS Hardware Offload with SR-IOV VF LAG, MCO needs to apply
linux bond configuration. It's important that SR-IOV Operator (installed day 2)
makes sure `switchdev` configurations are applied before linux bond
configuration, this probably requires a node rebooting to guarantee the order
if linux bond configuration is applied upfront.```